pygeos/predicates.py

```diff
@@ -650,14 +652,23 @@ def overlaps(a, b, **kwargs):
 
     Examples
     --------
-    >>> line = Geometry("LINESTRING(0 0, 1 1)")
-    >>> overlaps(line, line)
+    >>> poly = Geometry("POLYGON ((0 0, 0 4, 4 4, 4 0, 0 0))")
+    >>> overlaps(poly, poly)
+    False
+    >>> overlaps(poly, Geometry("POLYGON ((0 0, 0 2, 2 2, 2 0, 0 0))"))
     False
```

Comment:
I think we should put some hints here about the answer, like:

```suggestion
    A polygon that is fully contained does not overlap:

    >>> overlaps(poly, Geometry("POLYGON ((0 0, 0 2, 2 2, 2 0, 0 0))"))
    False
```

and maybe similar for some others that might not be directly clear